pkg/probe/checker.go

```diff
@@ -0,0 +1,75 @@
+// (C) Copyright Confidential Containers Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package probe
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type Checker struct {
+	Clientset        kubernetes.Interface
+	RuntimeclassName string
+}
+
+func (c *Checker) GetNodeName() string {
+	return os.Getenv("NODE_NAME")
+}
+
+func (c *Checker) GetAllPods(selector string) (result *corev1.PodList, err error) {
+	return c.Clientset.CoreV1().Pods("").List(context.TODO(), metav1.ListOptions{
+		FieldSelector: selector,
+	})
+}
+
+func (c *Checker) AllDedicatedPeerPodsAreReady() (ready bool, err error) {
```

Comment:
Any specific meaning behind using the term `Dedicated` ?

Reply:
@bpradipt no, we can change the func name for free.

Reply:
How about we rename it to GetAllPeerPods ?

Reply:
Sure, changed it @bpradipt